### PR TITLE
[CF-622] Validate if success file paths

### PR DESCRIPTION
**Pull request by ChandraAddala**
- Added success file paths to the conf file
 - Code to validate if the the success files exists and fail if they do not
 - Code to optionally skip validation of success file paths.

### CODE REVIEW

**Comment by ChandraAddala** [comment]
@shintasmith @usnavi @VinnyQ Pls review...

File: tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala [hunk 1]
```diff
@@ -60,30 +61,29 @@ class Archiver( runConfig : RunConfig ) {
     conf.getString("tiamat.identity.apiKey"),
     conf.getString("tiamat.identity.password"))
 
-  val token = identity.getToken()
+  lazy val token = identity.getToken()
```
Comment (shintasmith) [inline tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala]: What is the reason that you make these guys lazy?

Comment (ChandraAddala) [inline tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala]: 'val' makes an expression evaluate during initialization. Its making it difficult to test as I couldnt override those values with mock objects. Making it lazy would only get evaluated when its first referenced. So I was able to overide these 'val's with mock in my test object by making them lazy.

I havent read completely about it but looks like partial functions are a way to slove this kinda issue.

File: tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala [hunk 2]
```diff
@@ -102,8 +106,50 @@ class Archiver( runConfig : RunConfig ) {
...
+    val hadoopConf = spark.hadoopConfiguration
+    val fs = FileSystem.get(hadoopConf)
+
+    val successFilePathSet = getSuccessFilePaths
+    
+    val nonExistentSuccessFiles = successFilePathSet.filter(successFile => !fs.exists(new Path(successFile)))
+    
+    if (nonExistentSuccessFiles.size > 0) {
+      
+      val errorMessage: String = s"Missing success files :[${nonExistentSuccessFiles.mkString(",")}]"
```
Comment (usnavi) [inline tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala]: Can we add a Tiamat error code like here:  https://one.rackspace.com/display/cloudfeeds/Archiving+Monitoring+and+Alerting#ArchivingMonitoringandAlerting-Errorcodes

File: tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala [hunk 3]
```diff
@@ -102,8 +106,50 @@ class Archiver( runConfig : RunConfig ) {
...
+    val successFilePathTemplatesSet = conf.getStringList("tiamat.success.file.paths").toSet
+
+    (for {
+      region <- runConfig.regions
+      date <- runConfig.dates
+    } yield successFilePathTemplatesSet.map( filePathTemplate => {
+        filePathTemplate.replace("#REGION#", region)
+                        .replace("#RUNDATE#", dayFormat.print(date))
+      })).flatten.toSet
+  }
```
Comment (usnavi) [inline tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala]: if a path doesn't have the #REGION# and/or #RUNDATE# will we get duplicated entries in the list?  is that an issue?

Comment (usnavi) [inline tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala]: nm.  not an issue.

**Comment by usnavi** [comment]
other than the error code, lgtm

File: tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala (hunk 2 above)
Comment (ChandraAddala) [inline tiamat/src/main/scala/com/rackspace/feeds/archives/Archiver.scala]: @usnavi I had added the error code.

**Comment by usnavi** [comment]
thanks - lgtm

**Comment by shintasmith** [comment]
lgtm too

**Pull request merged**